### PR TITLE
Updated materials

### PR DESCRIPTION
**Pull request by nimarafati**
Hi Lucile! 

I updated the exercises and also uploaded the presentations. For presentations I included QR codes plus link to the page. 
It will be great if you can double check my changes.

Regards,

### CODE REVIEW

File: labs/functional_annotation.md [hunk 1]
```diff
@@ -18,15 +18,15 @@ Setup the folder structure:
...
+export data=/home/data/data_annotation/
 export functional_annotation_path=~/annotation/functional_annotation
 export structural_annotation_path=~/annotation/structural_annotation
 mkdir -p $functional_annotation_path
 ```
 You need to have the right to write in the folder blastdb/ for the next exercises so you will copy the folder:
 ```bash
-cp -r /home/data/byod/Annotation/data/blastdb .
-chmod +w blastdb/uniprot_dmel/
+chmod +w $data/blastdb/uniprot_dmel/
```
Comment (LucileSol) [inline labs/functional_annotation.md:30]: it should be something like 
cd $functional_annotation_path (to copy it folder in it)
cp -r $data/blastdb/uniprot_dmel/
chmod +w $functional_annotation_path/uniprot_dmel/

File: labs/functional_annotation.md [hunk 2]
```diff
@@ -86,10 +87,10 @@ This analyse will fail.
 If you did not have a look at the maker_final.faa, please have look and find a solution to
 make interproscan run.  
 
-:bulb:***TIP*** : ```gff3_sp_extract_sequences.pl --help```  
+:bulb:***TIP*** : ```agat_sp_extract_sequences.pl --help```  
 
 <details>
-<summary> :key:**Interproscan problem** - Click to expand the solution </summary>
+<summary> :key: **Interproscan problem** - Click to expand the solution </summary>
```
Comment (LucileSol) [inline labs/functional_annotation.md:95]: gff3_sp_extract_sequences.pl -g maker_final.gff -f genome.fa --cfs --cis -p -o maker_final_prot.fa 
should be 
agat_sp_extract_sequences.pl -g maker_final.gff -f genome.fa --cfs --cis -p -o maker_final_prot.fa 
in the answer as well

File: labs/functional_annotation.md [hunk 3]
```diff
@@ -59,7 +59,8 @@ ln -s $data/genome/genome.fa
 In order to do the functional annotation, you need to retrieve the proteins from the structural annotation.
 
 ```bash
-gff3_sp_extract_sequences.pl -g maker_final.gff -f genome.fa -p -o maker_final_prot.fa
+conda activate agat
+agat_sp_extract_sequences.pl -g maker_final.gff -f genome.fa -p -o maker_final_prot.fa
 ```
  <br />
```
Comment (LucileSol) [inline labs/functional_annotation.md:67]: prodom is not working, do you want to chose an other one?

File: labs/functional_annotation.md [hunk 4]
```diff
@@ -118,28 +119,38 @@ Next, you could write scripts of your own to merge interproscan output into your
...
+Blast searches provide an indication about potential homology to known proteins. This approach can be used to give a name to the genes and a function to the transcripts. A 'full' Blast analysis can run for several days and consume several GB of Ram. Consequently, for a huge amount of data it is recommended to parallelize this step doing analysis of chunks of tens or hundreds proteins. 
 
-### Perform Blast searches from the command line on Uppmax:
+### Perform Blast searches from the command line:
 
 To run Blast on your data, use the Ncbi Blast+ package against a Drosophila-specific database (included in the folder we have provided for you, under **~/annotation/blastdb/uniprot\_dmel/uniprot\_dmel.fa**) - of course, any other NCBI database would also work:
 ```
-/etc/ncbi-blast-2.9.0+-src/c++/ReleaseMT/bin/blastp -db ~/annotation/blastdb/uniprot_dmel/uniprot_dmel.fa -query maker_final_prot.fa -outfmt 6 -out blast.out -num_threads 4
+conda activate blast
+blastp -db ~/annotation/blastdb/uniprot_dmel/uniprot_dmel.fa -query maker_final_prot.fa -outfmt 6 -out blast.out -num_threads 4
```
Comment (LucileSol) [inline labs/functional_annotation.md]: correct path
blastp -db $functional_annotation_path/uniprot_dmel/uniprot_dmel.fa -query maker_final_prot.fa -outfmt 6 -out blast.out -num_threads 4

File: labs/functional_annotation.md [hunk 5]
```diff
@@ -118,28 +119,38 @@ Next, you could write scripts of your own to merge interproscan output into your
...
 ```
+if you received the following error:  
+_BLAST Database error: No alias or index file found for protein database [/blastdb/uniprot_dmel/uniprot_dmel.fa] in search path [/home/nima_rafati/functional_annotation::]_
+
+Then create the database again, it is due to new formating implemented in blast.  
+```
+cd $data/blastb/uniprot_dmel
```
Comment (LucileSol) [inline labs/functional_annotation.md:144]: I did not need to do it with the path I put before, if you want maybe you should write
$functional_annotation_path/uniprot_dmel

File: labs/functional_annotation.md [hunk 6]
```diff
@@ -118,28 +119,38 @@ Next, you could write scripts of your own to merge interproscan output into your
...
+```
+
 Against the Drosophila-specific database, the blast search takes about 2 secs per protein request - depending on how many sequences you have submitted, you can make a fairly deducted guess regarding the running time.
 
 ### load the retrieved information in your annotation file:  
 
 Now you should be able to use the following script:
 ```
-gff3_sp_manage_functional_annotation.pl -f maker_final.interpro/maker_final.gff -b blast.out --db ~/annotation/blastdb/uniprot_dmel/uniprot_dmel.fa -o maker_final.interpro.blast  
+agat_sp_manage_functional_annotation.pl -f maker_final.interpro/maker_final.gff -b blast.out --db ~/annotation/blastdb/uniprot_dmel/uniprot_dmel.fa -o maker_final.interpro.blast
```
Comment (LucileSol) [inline labs/functional_annotation.md:155]: add conda activate agat?

and still same path problem I think (correct me if I am wrong): 
agat_sp_manage_functional_annotation.pl -f maker_final.interpro/maker_final.gff -b blast.out --db ~/$functional_annotation_path/uniprot_dmel/uniprot_dmel.fa -o maker_final.interpro.blast

**Review by LucileSol — CHANGES_REQUESTED** [review]
please let me know if I miss something with the path!

File: labs/prokaryote_annotation.md [hunk 7]
```diff
@@ -15,7 +15,8 @@ Setup the folder structure:
 
 ```bash
 cd ~/annotation/
-export data=/home/data/byod/Annotation/data/
+conda activate bioinfo
+export data=/home/data/data_annotation/
 export bacterial_annotation_path=~/annotation/bacterial_annotation
 mkdir -p $bacterial_annotation_path
 export LC_ALL=C
```
Comment (LucileSol) [inline labs/prokaryote_annotation.md:22]: this part is not sued anymore : 
```
In order to use the blast functionality from within Prokka (see below) we need to edit the file "/home/export_paths/export_path.sh" and comment out this line:
```
export PATH=$PATH:/usr/local/WUBlast
```
the line will then look like this :
```
#export PATH=$PATH:/usr/local/WUBlast
```
After editing the file, please log out of the terminal and then log back in.
Execute the command blastn -version and inspect the output. It should be something like:
```
tblastn: 2.9.0+
Package: blast 2.9.0, build Jun 13 2019

File: labs/prokaryote_annotation.md [hunk 8]
```diff
@@ -15,7 +15,8 @@ Setup the folder structure:
 
 ```bash
 cd ~/annotation/
-export data=/home/data/byod/Annotation/data/
+conda activate bioinfo
```
Comment (LucileSol) [inline labs/prokaryote_annotation.md:18]: conda activate bioinfo not necessary but conda activate prokka yes

and later on to run busco (I can't comment there)
conda activate busco
busco -i prokka_Chlamydia/PROKKA_XXXX.faa -o chlamydia_busco_prot -m prot -c 8 -l bacteria_odb10

**Review by LucileSol — CHANGES_REQUESTED** [review]
there is some conda loading missing and the busco command line to change

File: labs/functional_annotation.md (hunk 1 above)
Comment (nimarafati) [reply labs/functional_annotation.md:30]: You are right.

File: labs/functional_annotation.md (hunk 2 above)
Comment (nimarafati) [reply labs/functional_annotation.md:95]: I had changed it to agat.

File: labs/functional_annotation.md (hunk 4 above)
Comment (nimarafati) [reply labs/functional_annotation.md]: I fixed the path

File: labs/functional_annotation.md (hunk 6 above)
Comment (nimarafati) [reply labs/functional_annotation.md:155]: You are absolutely right!.

**Review by nimarafati — COMMENTED** [review]
Thanks Lucile! 
Very valuable inputs :)

**Comment by nimarafati** [comment]
Hi Lucile,

I merged this PR after going through your suggestion on functional_annotation.

Thanks!

Regards,
Nima

**Pull request closed without merging**